src/sentry/services/hybrid_cloud/notifications/impl.py

```diff
@@ -162,3 +162,25 @@ def remove_notification_settings_for_user(
 
     def close(self) -> None:
         pass
+
+    def filter(
+        self,
+        *,
+        provider: Optional[ExternalProviders] = None,
+        type: Optional[NotificationSettingTypes] = None,
+        scope_type: Optional[NotificationScopeType] = None,
+        scope_identifier: Optional[int] = None,
+        user_ids: Optional[Iterable[int]] = None,
+        team_ids: Optional[Iterable[int]] = None,
```

Comment:
Should we be using the `filter` parameter with a TypedDict here like we do for [UserService.get_many](https://github.com/getsentry/sentry/blob/08131866a01716bd0f9019dfebe0ff30513e3a9f/src/sentry/services/hybrid_cloud/user/service.py#L45). We used a typed dict as it would be simpler to serialize into RPC calls.

Reply:
👀